huaweicloud/services/dli/resource_huaweicloud_dli_queue.go

```diff
@@ -357,6 +365,20 @@ func resourceDliQueueUpdate(ctx context.Context, d *schema.ResourceData, meta in
 		}
 	}
 
+	oldVal, newVal := d.GetChange("elastic_resource_pool_name")
+	if oldVal == "" && newVal != "" {
+		associateQueueToElasticResourcePoolOpts := elasticresourcepool.AssociateQueueOpts{
+			ElasticResourcePoolName: newVal.(string),
+			QueueName:               queueName,
+		}
+		err = associateQueueToElasticResourcePool(cfg, region, associateQueueToElasticResourcePoolOpts)
+		if err != nil {
+			return diag.Errorf("error associate queue to elastic resopurce pool: %s", err)
+		}
+	} else if oldVal != "" {
+		log.Printf("[WARN] the queue has been associate with an Elastic resopurce pool")
```

Comment:
we should return an error instead of adding a log message

Reply:
fixed